lib/entities/team-membership.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * TeamMembership instances
+ * @namespace TeamMembership
+ */
+
+import cloneDeep from 'lodash/cloneDeep'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import enhanceWithMethods from '../enhance-with-methods'
+import errorHandler from '../error-handler'
+
+/**
+ * @memberof TeamMembership
+ * @typedef TeamMembership
+ * @prop {Meta.Sys} sys - System metadata
+ * @prop {boolean} admin - User is an admin
+ * @prop {function(): Object} toPlainObject() - Returns this Team Membership as a plain JS object
+ */
+
+/**
+ * @memberof TeamMembership
+ * @typedef TeamMembershipCollection
+ * @prop {number} total
+ * @prop {number} limit
+ * @prop {number} skip
+ * @prop {Object<{type: "Array"}>} sys
+ * @prop {Array<TeamMembership.TeamMembership>} items
+ * @prop {function(): Object} toPlainObject() - Returns the collection as a plain JS object
+ */
+
+function createTeamMembershipApi (http, teamId) {
+  return {
+    /**
+     * Sends an update to the server with any changes made to the object's properties
+     * @memberof TeamMembership
+     * @func update
+     * @return {Promise<TeamMembership>} Object returned from the server with updated changes.
+     * @example
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getOrganizations()
+     * .then((organizations) => organizations.items[0])
+     * .then(org => org.getTeamMembership('teamId', 'teamMembership_id'))
+     * .then((teamMembership) => {
+     *  teamMembership.admin = true;
+     *  teamMembership.update();
+     * })
+     * .catch(console.error)
+     */
+
+    update: function () {
+      const raw = this.toPlainObject()
+
+      return http.put('teams/' + teamId + '/team_memberships/' + this.sys.id, raw, {
+        headers: {
+          'X-Contentful-Version': this.sys.version || 0
+        }
+      })
+        .then((response) => wrapTeamMembership(http, response.data), errorHandler)
+    },
+
+    /**
+     * Deletes this object on the server.
+     * @memberof TeamMembership
+     * @func delete
+     * @return {Promise} Promise for the deletion. It contains no data, but the Promise error case should be handled.
+     * @example
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getOrganizations()
+     * .then((organizations) => organizations.items[0])
+     * .then(org => org.getTeamMembership('teamId', 'teamMembership_id'))
+     * .then((teamMembership) => {
+     *  teamMembership.delete();
+     * })
+     * .catch(console.error)
+     */
+
+    delete: function () {
+      return http.delete('teams/' + teamId + '/team_memberships/' + this.sys.id,
+        {headers: {'x-contentful-enable-alpha-feature': 'teams-api'}}
```

Comment:
only delete needs the alpha header?

Reply:
yes, looks like the update does need it; in fact I will probably remove the alpha header in this pr too since was removed on GK side today

Reply:
just fyi, not everything was deployed on GK side, so I will remove the alpha header in my next task that is also related (the getOrganizationTeamMemberships endpoint)